sdk/highlight-node/src/hooks.ts

```diff
@@ -0,0 +1,65 @@
+// inspired by https://github.com/getsentry/sentry-javascript/issues/5639
+
+export function hookOutput(
+	writer: 'stdout' | 'stderr',
+	callback: (buffer: string) => void,
+) {
+	let old_write = process[writer].write
+
+	process[writer].write = (str: string, enc: any, cb?: any) => {
+		callback(str)
+		return old_write.apply(old_write, [str, enc, cb])
+	}
+
+	return () => {
+		process[writer].write = old_write
+	}
+}
+
+interface ConsolePayload {
+	date: Date
+	level: string
+	message: string
+	stack: object
+}
+
+type ConsoleFn = (...data: any) => void
+
+let consoleHooked = false
+
+export function hookConsole(cb: (cb: ConsolePayload) => void) {
+	if (consoleHooked) return
+	consoleHooked = true
+	const levels = [
+		'debug',
+		'info',
+		'log',
```

Comment:
Here's where we normalize the logs to the frontend:
https://github.com/highlight/highlight/blame/main/backend/clickhouse/logs.go#L202-L234

`log` will get normalized to `info` when it is shown the to the end user on the frontend. Hence, to find this log, they expect to search for `level:info`.

However, I think we should normalize this _before_ we persist to the database. The reason being is that if this is stored as `SeverityText=log` in the db, then we can't filter for it via `level:info`.

Realistically, every language/sdk is going to add these arbitrary levels  (other things in this list including `count`, `assert`, `dir`, etc.) and we need to land on some agreed standard across the board.

If we don't normalize here, can we normalize at the OTEL level (I don't think anything exists but correct me if I'm wrong).

Reply:
I just noticed that we have a `consoleMethodsToRecord` [option](https://www.highlight.io/docs/sdk/client#consoleMethodsToRecord). We should respect that option when deciding what to monkeypatch.

Reply:
great point on both! making those improvements now

Reply:
using the `consoleMethodsToRecord` option now when set, and mapping the levels to their go parsed values.